src/connectors/autocomplete/connectAutocomplete.ts

```diff
@@ -0,0 +1,171 @@
+  name: 'autocomplete',
+  connector: true,
+});
+
+interface AutocompleteIndex {
+  index: string;
+  hits: Hits;
+  results: SearchResults | undefined;
+}
+
+interface AutocompleteConnectorParams {
+  /**
+   * Escapes HTML entities from hits string values.
+   *
+   * @default `true`
+   */
+  escapeHTML?: boolean;
+}
+
+type RefineFunction = (query: string) => Helper;
+
+export interface AutocompleteRendererOptions<TAutocompleteWidgetParams>
+  extends RendererOptions<TAutocompleteWidgetParams> {
+  currentRefinement: string;
+  indices: AutocompleteIndex[];
+  instantSearchInstance: InstantSearch;
+  refine: RefineFunction;
+}
+
+export type AutocompleteRenderer<TAutocompleteWidgetParams> = Renderer<
+  AutocompleteRendererOptions<
+    AutocompleteConnectorParams & TAutocompleteWidgetParams
+  >
+>;
+
+export type AutocompleteWidgetFactory<
+  TAutocompleteWidgetParams
+> = WidgetFactory<AutocompleteConnectorParams & TAutocompleteWidgetParams>;
+
+export type AutocompleteConnector = <TAutocompleteWidgetParams>(
+  render: AutocompleteRenderer<TAutocompleteWidgetParams>,
+  unmount?: Unmounter
+) => AutocompleteWidgetFactory<TAutocompleteWidgetParams>;
+
+const connectAutocomplete: AutocompleteConnector = (
+  renderFn,
+  unmountFn = noop
+) => {
+  checkRendering(renderFn, withUsage());
+
+  return widgetParams => {
+    const { escapeHTML = true } = widgetParams || {};
+
+    warning(
+      !(widgetParams as any).indices,
+      `Since InstantSearch.js 4, \`connectAutocomplete\` infers the indices from the tree of widgets.
+The \`indices\` option is ignored.`
+    );
+
+    type ConnectorState = {
+      instantSearchInstance?: InstantSearch;
+      refine?: RefineFunction;
+    };
+
+    const connectorState: ConnectorState = {};
+
+    return {
+      $$type: 'ais.autocomplete',
+
+      getConfiguration() {
+        const parameters = {
+          query: '',
+        };
+
+        if (!escapeHTML) {
+          return parameters;
+        }
+
+        return {
+          ...parameters,
+          ...TAG_PLACEHOLDER,
+        };
+      },
+
+      init({ instantSearchInstance, helper }) {
+        connectorState.instantSearchInstance = instantSearchInstance;
+        connectorState.refine = (query: string) =>
+          helper.setQuery(query).search();
+
+        renderFn(
+          {
+            widgetParams,
+            currentRefinement: helper.state.query || '',
+            indices: [],
+            refine: connectorState.refine,
+            instantSearchInstance: connectorState.instantSearchInstance,
+          },
+          true
+        );
+      },
+
+      render({ helper, scopedResults }) {
+        const indices = scopedResults.map(scopedResult => {
+          scopedResult.results.hits = escapeHTML
+            ? escapeHits(scopedResult.results.hits)
+            : scopedResult.results.hits;
+
+          return {
+            index: scopedResult.results.index,
+            hits: scopedResult.results.hits,
```

Comment:
something which the previous version also doesn't implement yet, and thus could be done in a separate PR, is adding `queryID` & absolute position to the hits array so it can be used for insights, like it can for Hits & InfiniteHits; cc @tkrugg 

see also https://github.com/algolia/react-instantsearch/pull/2687